crates/sui-json-rpc/src/axum_router.rs

```diff
@@ -0,0 +1,426 @@
+                ))
+            }
+            Some((name, method)) => match method.inner() {
+                MethodKind::Sync(callback) => {
+                    logger.on_call(
+                        name,
+                        params.clone(),
+                        logger::MethodKind::MethodCall,
+                        TransportProtocol::Http,
+                    );
+                    Some((callback)(id, params, max_response_body_size as usize))
+                }
+                MethodKind::Async(callback) => {
+                    logger.on_call(
+                        name,
+                        params.clone(),
+                        logger::MethodKind::MethodCall,
+                        TransportProtocol::Http,
+                    );
+
+                    let id = id.into_owned();
+                    let params = params.into_owned();
+
+                    Some(
+                        (callback)(id, params, conn_id, max_response_body_size as usize, None)
+                            .await,
+                    )
+                }
+
+                MethodKind::Subscription(callback) => {
+                    logger.on_call(
+                        name,
+                        params.clone(),
+                        logger::MethodKind::Subscription,
+                        TransportProtocol::WebSocket,
+                    );
+                    if let Some(cn) = bounded_subscriptions.acquire() {
+                        let conn_state = ConnState {
+                            conn_id,
+                            close_notify: cn,
+                            id_provider,
+                        };
+                        println!("hello");
+                        callback(id.clone(), params, sink.clone(), conn_state, None).await;
+                        None
+                    } else {
+                        Some(MethodResponse::error(
+                            id,
+                            reject_too_many_subscriptions(bounded_subscriptions.max()),
+                        ))
+                    }
+                }
+
+                MethodKind::Unsubscription(callback) => {
+                    logger.on_call(
+                        name,
+                        params.clone(),
+                        logger::MethodKind::Unsubscription,
+                        TransportProtocol::WebSocket,
+                    );
+
+                    Some(callback(
+                        id,
+                        params,
+                        conn_id,
+                        max_response_body_size as usize,
+                    ))
+                }
+            },
+        };
+
+        if let Some(response) = &response {
+            logger.on_result(
+                name,
+                response.success,
+                response.error_code,
+                request_start,
+                TransportProtocol::Http,
```

Comment:
probs noob q: do we need this/ how important is it to have this be TransportProtocol::WebSocket for MethodKind::(Un)subscription?

Reply:
oh, nice catch, i can update that...although afaik its not really used for anything.